hazelcast/src/main/java/com/hazelcast/scheduledexecutor/IScheduledExecutorService.java

```diff
@@ -0,0 +1,395 @@
+     * after the given delay at the given {@link Member}.
+     *
+     * @param command the task to execute
+     * @param member the member to execute the task
+     * @param delay the time from now to delay execution
+     * @param unit the time unit of the delay parameter
+     * @return a ScheduledFuture representing pending completion of
+     *         the task and whose {@code get()} method will return
+     *         {@code null} upon completion
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    IScheduledFuture<?> scheduleOnMember(Runnable command, Member member, long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a one-shot action that becomes enabled
+     * after the given delay at the given {@link Member}.
+     *
+     * @param command the task to execute
+     * @param member the member to execute the task
+     * @param delay the time from now to delay execution
+     * @param unit the time unit of the delay parameter
+     * @return a ScheduledFuture representing pending completion of
+     *         the task and whose {@code get()} method will return
+     *         {@code null} upon completion
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    <V> IScheduledFuture<V> scheduleOnMember(Callable<V> command, Member member, long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a periodic action that becomes enabled first
+     * after the given initial delay, and subsequently with the given
+     * period at the given {@link Member}. Executions will commence after
+     * {@code initialDelay} then {@code initialDelay+period}, then
+     * {@code initialDelay + 2 * period}, and so on.
+     * If any execution of this task
+     * takes longer than its period, then subsequent execution will be skipped.
+     *
+     * @param command the task to execute
+     * @param member the member to execute the task
+     * @param initialDelay the time to delay first execution
+     * @param period the period between successive executions
+     * @param unit the time unit of the initialDelay and period parameters
+     * @return a ScheduledFuture representing pending completion of
+     *         the task, and whose {@code get()} method will throw an
+     *         exception upon cancellation
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    IScheduledFuture<?> scheduleOnMemberWithRepetition(Runnable command, Member member,
+                                                       long initialDelay, long period, TimeUnit unit);
+
+    /**
+     * Creates and executes a one-shot action that becomes enabled
```

Comment:
What happens when the partition is migrated before the task is started, or during task execution?